File: demos/demo_lightblue_segmenter.py
#!/usr/bin/env python3
"""Run the 'lightblue' segmenter on single images"""
import argparse
import sys

import cv2
import numpy as np
from trifinger_object_tracking.py_lightblue_segmenter import segment_image


def main():
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("input_image")
    args = parser.parse_args()

    # load the image
    img = cv2.imread(args.input_image)

    # get the segmentation mask
    mask = segment_image(img)

    # for nicer visualisation get the masked area from the original image
    segmented = cv2.bitwise_and(img, img, mask=mask)

    # stack together for visualisation
    both = np.hstack([img, segmented])
    cv2.imshow("Lightblue Segmenter", both)
    cv2.waitKey()

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: demos/demo_lightblue_segmenter.cpp
#include <iostream>
#include <string>
#include <opencv2/opencv.hpp>
#include <trifinger_object_tracking/xgboost_classifier_single_color_rgb.h>

int main(int argc, char *argv[])
{
    if (argc != 2)
    {
        std::cout << "Wrong number of arguments. " << argc << std::endl;
        std::cout << "Usage: " << argv[0] << " path/to/image" << std::endl;
        return 1;
    }

    std::string image_path(argv[1]);
    std::cout << "load image " << image_path << std::endl;

    cv::Mat img = cv::imread(image_path);
    cv::Mat mask = trifinger_object_tracking::segment_image(img);

    // apply the mask on the image to get the segmented areas
    cv::Mat segmented;
    img.copyTo(segmented, mask);

    // concat images for nicer visualisation
    cv::Mat merged;
    cv::hconcat(img, segmented, merged);
    cv::imshow("Lightblue Segmenter", merged);
    cv::waitKey(0);

    return 0;
}
